### PR TITLE
add rule for mediaHasCaption and tests

### PR DESCRIPTION
**Pull request by VictorHom**
[rule](https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/master/docs/rules/media-has-caption.md)

from spreadsheet #341

### CODE REVIEW

File: packages/@romejs/js-compiler/lint/rules/jsx-a11y/mediaHasCaption.test.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+		await testLintMultiple(
+			t,
+			[
+				// INVALID
+				"<audio {...props} />",
+				"<video {...props} />",
+				// // VALID
+				"<audio><track kind='captions' {...props} /></audio>",
+				"<video><track kind='captions' {...props} /></video>",
+				"<video muted {...props} ></video>",
```
Comment (bitpshr) [inline packages/@romejs/js-compiler/lint/rules/jsx-a11y/mediaHasCaption.test.ts:16]: Can we also add a test case for a muted `audio` element?

Comment (VictorHom) [reply packages/@romejs/js-compiler/lint/rules/jsx-a11y/mediaHasCaption.test.ts:16]: just noticed my `// // VALID` which I'll fix.

I think audio shouldn't have a muted property, at least from how I read the [rule ](https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/master/docs/rules/media-has-caption.md)

**Comment by VictorHom** [comment]
caught up with master to fix the conflicts

**Comment by sebmck** [comment]
Sweet! Looks great. Thank you! I was going to manually merge and rebase but misinterpreted the comments above to mean there was some pending changes.

**Pull request merged**